adapters/httpapi_winhttp.c

```diff
@@ -736,6 +736,11 @@ HTTPAPI_RESULT HTTPAPI_SetOption(HTTP_HANDLE handle, const char* optionName, con
             /*winhttp accepts all certificates, because it actually relies on the system ones*/
             result = HTTPAPI_OK;
         }
+        else if (strcmp(OPTION_HTTP_PROXY, optionName) == 0)
+        {
+            /*winhttp accepts all proxy, because it actually relies on the system ones*/
+            result = HTTPAPI_OK;
+        }
```

Comment:
Please change this to error out and produce a log message stating that comment you have. The reason we need this to error out so that it is not misleading for users of the SDK thinking that the options they provided thru the SDK are in effect, while the options are set in OS.

Reply:
Or simply remove the check and let it fall thru the next block.

---
In reply to: [150671536](https://github.com/Azure/azure-c-shared-utility/pull/126#discussion_r150671536) [](ancestors = 150671536)